ui-tests/src/test/java/plugins/WarningsNextGenerationPluginTest.java

```diff
@@ -0,0 +1,621 @@
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.DumbSlave;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.Job;
+import org.jenkinsci.test.acceptance.po.Slave;
+import org.jenkinsci.test.acceptance.po.WorkflowJob;
+
+import static org.jenkinsci.test.acceptance.plugins.warnings_ng.Assertions.*;
+
+/**
+ * Acceptance tests for the Warnings Next Generation Plugin.
+ *
+ * @author Frank Christian Geyer
+ * @author Ullrich Hafner
+ * @author Manuel Hampp
+ * @author Anna-Maria Hardi
+ * @author Elvira Hauer
+ * @author Deniz Mardin
+ * @author Stephan Plöderl
+ * @author Alexander Praegla
+ * @author Michaela Reitschuster
+ * @author Arne Schöntag
+ * @author Alexandra Wenzel
+ * @author Nikolai Wohlgemuth
+ * @author Florian Hageneder
+ * @author Veronika Zwickenpflug
+ */
+@WithPlugins("warnings-ng")
+public class WarningsNextGenerationPluginTest extends AbstractJUnitTest {
+    private static final String WARNINGS_PLUGIN_PREFIX = "/warnings_ng_plugin/";
+
+    private static final String CHECKSTYLE_ID = "checkstyle";
+    private static final String ANALYSIS_ID = "analysis";
+    private static final String CPD_ID = "cpd";
+    private static final String PMD_ID = "pmd";
+    private static final String FINDBUGS_ID = "findbugs";
+    private static final String MAVEN_ID = "maven-warnings";
+
+    private static final String WARNING_LOW_PRIORITY = "Low";
+
+    private static final String SOURCE_VIEW_FOLDER = WARNINGS_PLUGIN_PREFIX + "source-view/";
+
+    private static final String CPD_SOURCE_NAME = "Main.java";
+    private static final String CPD_SOURCE_PATH = "duplicate_code/Main.java";
+
+    /**
+     * Credentials to access the docker container. The credentials are stored with the specified ID and use the provided
+     * SSH key. Use the following annotation on your test case to use the specified docker container as git server or
+     * build agent:
+     * <blockquote>
+     * <pre>@Test @WithDocker @WithCredentials(credentialType = WithCredentials.SSH_USERNAME_PRIVATE_KEY,
+     *                                    values = {CREDENTIALS_ID, CREDENTIALS_KEY})}
+     * public void shouldTestWithDocker() {
+     * }
+     * </pre></blockquote>
+     */
+    private static final String CREDENTIALS_ID = "git";
+    private static final String CREDENTIALS_KEY = "/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/unsafe";
+
+    @Inject
+    private DockerContainerHolder<JavaGitContainer> dockerContainer;
+
+    /**
+     * Runs a pipeline with all tools two times. Verifies the analysis results in several views. Additionally, verifies
+     * the expansion of tokens with the token-macro plugin.
+     */
+    @Test
+    @WithPlugins({"token-macro", "pipeline-stage-step", "workflow-durable-task-step", "workflow-basic-steps"})
+    public void should_record_issues_in_pipeline_and_expand_tokens() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'should_record_issues_in_pipeline_and_expand_tokens' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5070362)